lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+import 'dart:async';
+import 'dart:collection';
+import 'package:brunstadtv_app/graphql/client.dart';
+import 'package:brunstadtv_app/graphql/queries/calendar.graphql.dart';
+import 'package:brunstadtv_app/router/router.gr.dart';
+import 'package:flutter_riverpod/src/consumer.dart'
+    show ConsumerState, ConsumerStatefulWidget;
+import 'package:flutter/material.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:riverpod/riverpod.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:intl/intl.dart';
+
+class CalendarPage extends ConsumerStatefulWidget {
+  const CalendarPage({super.key});
+
+  @override
+  ConsumerState<CalendarPage> createState() => _CalendarPageState();
+}
+
+class _CalendarPageState extends ConsumerState<CalendarPage> {
+  DateTime _focusedDay = DateTime.now();
+  DateTime? _selectedDay = DateTime.now();
+
+  bool _isCalendarCollapse = true;
+
+  Future<Query$CalendarPeriod$calendar$period?>? _calendarPeriod;
+  Future<Query$CalendarDay$calendar?>? _selectedDayEvent = null;
+
+  HashSet<DateTime> activeDaysPeriod = HashSet<DateTime>();
+  List<String> eventsPeriod = [];
+
+  final kFirstDay = DateTime(
```

Comment:
Do we have to set first and last day? Can you set it to 5 million years in the future and past?

Reply:
5 billion seems too much, I just cut it to 5000 year front and back ward

Reply:
But why do we have to set it at all?

Reply:
because the table calendar component required it